### ISSUE TITLE
Win10 May 2020 Update

### ISSUE THREAD

**Issue by Disk2019** [undergroundwires/privacy.sexy#10]
First of all i want to thank you Brother for Your Script which is awesome & does all the tweaks for telemetery as well as debloating online Win10.
Since May 2020 Update M$ has changed some store apps name like cortana & many more.
Please update your script to fully support May 2020 Update of Win 10.

**Comment by Disk2019** [comment]
Brother i have found errors in your script where HKU registry values are found which will only work if script will first load ntuser.dat into registry to apply the tweak but theres nothing like that hence failed to tweak those registry values. 
then script have to unload the same after applied hku tweaks.
i hope it will be resolved in your next update as well.

**Comment by Disk2019** [comment]
Brother please do update your script as well to enable Dism.exe /online /Cleanup-Image /StartComponentCleanup /ResetBase via registry tweak into your script as it is disabled by default in win10.

**Comment by undergroundwires** [comment]
Hi @Disk2019 , thank you for your kind words & glad to hear some one else finding this tool useful. Thanks for the constructive feedback as well 🙏 Should  fix them all.

Let's take one by one:

> May 2020 adoptation

Thanks for the update 👍 I'll take look into May 2020 update when I have time to adopt appx uninstalls.

> HKU registry values

That's some gold worth bug report. 👏 Thanks.. Can we just change "HKU\DefaultUser" lines to "HKCU" instead? Then we don't need to run `reg load` & `reg unload`. [According to Wikipedia](https://en.wikipedia.org/wiki/Windows_Registry#HKEY_USERS_(HKU)): HKU contains subkeys corresponding to the HKEY_CURRENT_USER keys for each user profile actively loaded on the machine. So we can have simpler convention with just having sticking to HKEY_CURRENT_USER e.g. HKCU? What do you think about that?

> Script examples in HKU comment

You do inspiring stuff that does not exist in the tool. I'd want to use them too !! The privacy related ones feel like they should be in the tool:

- Gaming Disable background recording
- Turn off "Get tips, tricks and suggestions as you use Windows"
- Turn Off App Suggestions on Start in Windows 10 aka Disable Start Menu App Suggestions
- Disable Sync Provider Notifications
- additional disable cortana (can be added under existing cortana tweak)
- Disallow WebSearch in search bar

I'm open to your suggestion to add more to existing scripts or to create more tweaks. 

> Dism.exe /online /Cleanup-Image /StartComponentCleanup /ResetBase 

It's actually a very good tweak that I'd want to use myself.

--------------------------

🌟 Would you like to add those tweaks to the tool yourself? It'd be great with your contribution. You'll just need to edit a single file and create a pull request for it. I'll review it and then when it's merged it'll go to the website. You'll only need to change [this file](https://github.com/undergroundwires/privacy.sexy/blob/master/src/application/application.yaml) to change/add/remove any improvements. It has a simple tree structure, I'm sure you'd figure it out & feel free to ask if you need help with anything.

Example for DISM.exe tweak, I'd add it somewhere under "Privacy cleanup". Maybe on [Line 271](https://github.com/undergroundwires/privacy.sexy/blob/master/src/application/application.yaml#L271).

**Comment by undergroundwires** [comment]
Hi @Disk2019 , thank you for your kind words & glad to hear some one else finding this tool useful. Thanks for the constructive feedback as well 🙏 Should  fix them all.

Let's take one by one:

> May 2020 adoptation

Thanks for the update 👍 I'll take look into May 2020 update when I have time to adopt appx uninstalls.

> HKU registry values

That's some gold worth bug report. 👏 Thanks.. Can we just change "HKU\DefaultUser" lines to "HKCU" instead? Then we don't need to run `reg load` & `reg unload`. [According to Wikipedia](https://en.wikipedia.org/wiki/Windows_Registry#HKEY_USERS_(HKU)): HKU contains subkeys corresponding to the HKEY_CURRENT_USER keys for each user profile actively loaded on the machine. So we can have simpler convention with just having sticking to HKEY_CURRENT_USER e.g. HKCU? What do you think about that?

> Script examples in HKU comment

You do inspiring stuff that does not exist in the tool. I'd want to use them too !! The privacy related ones feel like they should be in the tool:

- Gaming Disable background recording
- Turn off "Get tips, tricks and suggestions as you use Windows"
- Turn Off App Suggestions on Start in Windows 10 aka Disable Start Menu App Suggestions
- Disable Sync Provider Notifications
- additional disable cortana (can be added under existing cortana tweak)
- Disallow WebSearch in search bar

I'm open to your suggestion to add more to existing scripts or to create more tweaks. 

> Dism.exe /online /Cleanup-Image /StartComponentCleanup /ResetBase 

It's actually a very good tweak that I'd want to use myself.

--------------------------

🌟 Would you like to add those tweaks to the tool yourself? It'd be great with your contribution. You'll just need to edit a single file and create a pull request for it. I'll review it and then when it's merged it'll go to the website. You'll only need to change [this file](https://github.com/undergroundwires/privacy.sexy/blob/master/src/application/application.yaml) to change/add/remove any improvements. It has a simple tree structure, I'm sure you'd figure it out & feel free to ask if you need help with anything.

Example for DISM.exe tweak, I'd add it somewhere under "Privacy cleanup". Maybe on [Line 271](https://github.com/undergroundwires/privacy.sexy/blob/master/src/application/application.yaml#L271).

Cheers 🍻

**Comment by Disk2019** [comment]
Can we just change "HKU\DefaultUser" lines to "HKCU" instead? Then we don't need to run reg load & reg unload. According to Wikipedia: HKU contains subkeys corresponding to the HKEY_CURRENT_USER keys for each user profile actively loaded on the machine. So we can have simpler convention with just having sticking to HKEY_CURRENT_USER e.g. HKCU? What do you think about that?

Reply : Brother i think yes you can but i am not that much perfect to suggest about the same. i have only edited your domain generated privacy script & set echo off to echo on to cross check where it fails on my win10 19041.329 VM Guest OS then searched for its countermeasures to be applied so that script works fine.

    Script examples in HKU comment

You do inspiring stuff that does not exist in the tool. I'd want to use them too !! The privacy related ones feel like they should be in the tool:

    Gaming Disable background recording
    Turn off "Get tips, tricks and suggestions as you use Windows"
    Turn Off App Suggestions on Start in Windows 10 aka Disable Start Menu App Suggestions
    Disable Sync Provider Notifications
    additional disable cortana (can be added under existing cortana tweak)
    Disallow WebSearch in search bar

Brother to be very fare being trustworthy online as i never want to take credits of experts who did a great scripting job . example are not mine as its taken from https://gist.github.com/Carm01/3b83d78a0d99cc78ff1d46385ba68a13 profile script.

Thanks & Regards.

**Comment by Disk2019** [comment]
Brother i have updated dism resetbase tweak on line 271 as per suggested by you as well as updated HKU default user values to correct HKCU value so that loading & unloading of ntuser.dat isnt used as well as removed last experimental tweak which doesnt do anything rather first unload the HKU default user hive which is already loaded thern load it as its incorrect way i found. Kindly cross check my edit & approve them if found in a correct way . i will again edit & update the yaml file with extra tweaks after this one is approved & havent found any wrong edit.

Thanks & Regards.

**Comment by undergroundwires** [comment]
@Disk2019 bro thank you for all of your contributions really. One (#12) of them got through and should be deployed on production soon (I noticed an issue with deployment, working on it now). Scripts, fixes and the logic looks great. Just there are some indentation issues in the file as I commented #13  and #11 . 

Congratulations by the way as being the first code contributor in the project after me :)

I'd gladly add you as the main page as contributor but respect that you choose to not to be there. Very humble of you.

**Comment by undergroundwires** [comment]
@Disk2019 I fixed the issues and merged the changes manually. I also solved the the deployment problem I mentioned (#14) and they are all deployed to production now. Feel free to do more changes or close the issue if you feel like we're done.

**Comment by undergroundwires** [comment]
@Disk2019 I fixed the issues and merged the changes manually. I also solved the the deployment problem I mentioned (#14) and they are all deployed to production now. Here are the [changelogs](
https://github.com/undergroundwires/privacy.sexy/releases/tag/0.4.6).

Happy to add some new useful tweaks & fix problematic ones including `HKU` stuff.. We have not resolved the issue with May 2020 update though.

**Comment by undergroundwires** [comment]
@Disk2019 I fixed the issues and merged the changes manually. I also solved the the deployment problem I mentioned (#14) and they are all deployed to production now. Here are the changelogs.

Happy to add some new useful tweaks & fix problematic ones including HKU stuff.. We have not resolved the issue with May 2020 update though. The documentation I used to get app names for scripts were taken [from here](https://docs.microsoft.com/en-us/windows/application-management/apps-in-windows-10). It's not updated with new stuff. I wonder where we can find documentation about new app names 🤔. 

Otherwise feel free to close the issue if you feel like we're done. Thanks for the effort & very good job btw, I'm sure community would appreciate that.
